src/components/ConnectionConfig.tsx

```diff
@@ -0,0 +1,307 @@
+
+const DS_TYPES_THAT_SUPPORT_TEMP_CREDS = ['cloudwatch'];
+const toOption = (value: string) => ({ value, label: value });
+const isAwsAuthType = (value: any): value is AwsAuthType => {
+  return typeof value === 'string' && awsAuthProviderOptions.some((opt) => opt.value === value);
+};
+export interface ConnectionConfigProps<
+  J extends AwsAuthDataSourceJsonData = AwsAuthDataSourceJsonData,
+  S = AwsAuthDataSourceSecureJsonData
+> extends DataSourcePluginOptionsEditorProps<J, S> {
+  standardRegions?: string[];
+  loadRegions?: () => Promise<string[]>;
+  defaultEndpoint?: string;
+  skipHeader?: boolean;
+  skipEndpoint?: boolean;
+  children?: React.ReactNode;
+  labelWidth?: number;
+  inExperimentalAuthComponent?: boolean;
+  externalId?: string;
+}
+
+export const ConnectionConfig: FC<ConnectionConfigProps> = (props: ConnectionConfigProps) => {
+  const [isARNInstructionsOpen, setIsARNInstructionsOpen] = useState(false);
+  const [regions, setRegions] = useState((props.standardRegions || standardRegions).map(toOption));
+  const { loadRegions, onOptionsChange, skipHeader = false, skipEndpoint = false } = props;
+  const { options } = props;
+  let profile = options.jsonData.profile;
+  if (profile === undefined) {
+    profile = options.database;
+  }
+  const tempCredsFeatureEnabled =
+    config.featureToggles.awsDatasourcesTempCredentials && DS_TYPES_THAT_SUPPORT_TEMP_CREDS.includes(options.type);
+  const awsAssumeRoleEnabled = config.awsAssumeRoleEnabled ?? true;
+  const awsAllowedAuthProviders = useMemo(
+    () =>
+      config.awsAllowedAuthProviders
+        .filter((provider) => (provider === AwsAuthType.GrafanaAssumeRole ? tempCredsFeatureEnabled : true))
+        .filter(isAwsAuthType),
+    [tempCredsFeatureEnabled]
+  );
+
+  const currentProvider = awsAuthProviderOptions.find((p) => p.value === options.jsonData.authType);
+
+  useEffect(() => {
+    // Make sure a authType exists in the current model
+    if (!currentProvider && awsAllowedAuthProviders.length) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          authType: awsAllowedAuthProviders[0],
+        },
+      });
+    }
+  }, [currentProvider, options, onOptionsChange, awsAllowedAuthProviders]);
+
+  useEffect(() => {
+    if (!loadRegions) {
+      return;
+    }
+
+    loadRegions().then((regions) => setRegions(regions.map(toOption)));
+  }, [loadRegions]);
+
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div data-testid="connection-config">
+      <ConfigSection title={skipHeader ? '' : 'Connection Details'} data-testid="connection-config">
+        <ConfigSubSection title="Authentication">
+          <Field
+            label="Authentication Provider"
+            description="Specify which AWS credentials chain to use."
+            htmlFor="authProvider"
+          >
+            <Select
+              id="authProvider"
+              aria-label="Authentication Provider"
```

Comment:
I'm curious is the aria-label here necessary/recommended by our accessibility folks? I would guess that the label + htmlFor would do the job of labeling this component but I could be wrong, I know these Select components are very tricky sometimes

Reply:
Ok so I tested the aria-labels with a screen reader and it looks like we need them on Select components, but not on the new Inputs. I would expect the aria labels be applied if we have an htmlFor and id, but they aren't getting. I will open a bug ticket for this in main grafana, but in the meantime I removed new <Field/> Input aria-labels but keeping the Select ones. I didn't touch the old ones, since they will be removed later anyway. Good catch!